### PR TITLE
api to UI slice included in client project, needs evolving but basic …

### PR DESCRIPTION
**Pull request by ohyde**
…steps are avilable now

Putting this in to explain different steps to kels later on :)

### CODE REVIEW

**Comment by ohyde** [comment]
Things to read about here Kels:

- redux
- actions (part of redux)
- reducers (part of redux)
- sagas ( don't need to understand, I don't really anyway :P )

I'll walk you through the flow on skype sometime soon.

File: client/src/api/invoices.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export default class ApiInvoices {
+  static fetch() {
+    return new Promise(resolve => {
```
Comment (lucasschejtman) [inline client/src/api/invoices.js]: no love for async/await ?

File: client/src/common/configs.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  DEVELOPMENT: {
+    incomeAPI: 'http://localhost:3000/income' 
+  },
+  PRODUCTION: {
+    incomeAPI: 'http://ezm.com/income'
+  }
+};
+
+// ENVIRONMENT retrieved from webpack
+export const CONFIG = R.prop(ENVIRONMENT, configMap);
```
Comment (lucasschejtman) [inline client/src/common/configs.js]: I think these sort of webpack globals should be more easily identifiable like ```__ENVIRONMENT__``` or something like that

Comment (lucasschejtman) [inline client/src/common/configs.js]: Also, I'm a bit meh wrt config in code but we can change that later on

File: client/src/containers/home/index.js [hunk 3]
```diff
@@ -12,7 +15,13 @@ const Input = styled.input`
 	border-radius: 3px;
 `;
 
-export default class Home extends React.Component {
+class Home extends React.Component {
+  constructor (props) {
+    super(props);
+
+    props.invoicesFetch();
```
Comment (lucasschejtman) [inline client/src/containers/home/index.js]: I think this goes on componentMount

File: client/src/containers/home/index.js [hunk 4]
```diff
@@ -25,3 +34,24 @@ export default class Home extends React.Component {
...
+};
+
+const mapStateToProps = (state) => {
+  console.log(state);
+  return {};
+};
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    invoicesFetch: () => { dispatch(invoicesFetch()); },
```
Comment (lucasschejtman) [inline client/src/containers/home/index.js]: don't think you need the ```{ }``` here

File: client/src/reducers/invoices/invoicesReducer.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import R from 'ramda';
+import { invoicesFetchSuccess } from 'actions/invoices';
+
+const defaultState = {
+	invoices: []
```
Comment (lucasschejtman) [inline client/src/reducers/invoices/invoicesReducer.js]: mixed tabs/spaces on this file

File: client/src/sagas/index.js [hunk 6]
```diff
@@ -1,4 +1,8 @@
-// main saga generators
-export function* sagas() {
-  //yield [];
-}
+import { all, fork } from 'redux-saga/effects'
+import { invoicesSagas } from "sagas/invoices";
+
+export function* rootSaga() {
+	yield all([
```
Comment (lucasschejtman) [inline client/src/sagas/index.js]: spaces

File: client/src/api/invoices.js (hunk 1 above)
Comment (ohyde) [inline client/src/api/invoices.js]: Sure, but for demonstration purposes, I think better to build into them.

File: client/src/common/configs.js (hunk 2 above)
Comment (ohyde) [inline client/src/common/configs.js]: Sure

**Comment by ohyde** [comment]
ahahah I can't believe you can dismiss a review! what a cool feature haha

**Comment by lucasschejtman** [comment]
This is the first time I do a github review over old school comment based PR, not bad

**Pull request merged**